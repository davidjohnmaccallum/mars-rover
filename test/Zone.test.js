const Zone = require("../src/Zone");

const mockCanvas = {
  height: 0,
  width: 0, 
  textAlign: "",
  textBaseline: "",
  font: "",
  getContext: () => ({
    beginPath: () => {},
    moveTo: () => {},
    lineTo: () => {},
    closePath: () => {},
    stroke: () => {},
    fillText: () => {},
  })
}

test("Zone grid coordinates converted to a physical location correctly", () => {
  const zone1 = Zone({ ...mockCanvas, height: 600, width: 600 }, 12, 12);
  expect(zone1.getPhysicalLocation(0, 0)).toMatchSnapshot();
  expect(zone1.getPhysicalLocation(1, 1)).toMatchSnapshot();
  expect(zone1.getPhysicalLocation(12, 12)).toMatchSnapshot();
  expect(zone1.getPhysicalLocation(3, 5)).toMatchSnapshot();

  const zone2 = Zone({ ...mockCanvas, height: 800, width: 500 }, 10, 5);
  expect(zone2.getPhysicalLocation(0, 0)).toMatchSnapshot();
  expect(zone2.getPhysicalLocation(1, 1)).toMatchSnapshot();
  expect(zone2.getPhysicalLocation(3, 5)).toMatchSnapshot();
});

test("Zone grid coordinates converted to a physical location correctly. Error handling.", () => {
  const zone1 = Zone({ ...mockCanvas, height: 600, width: 600 }, 12, 12);
  expect(zone1.getPhysicalLocation(-3, 5)).toMatchSnapshot();

  const zone2 = Zone({ ...mockCanvas, height: 800, width: 500 }, 10, 5);
  expect(zone2.getPhysicalLocation(12, 12)).toMatchSnapshot();
  expect(zone2.getPhysicalLocation(-3, 5)).toMatchSnapshot();
});
