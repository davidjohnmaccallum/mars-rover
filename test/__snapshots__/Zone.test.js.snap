// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zone grid coordinates converted to a physical location correctly 1`] = `
Object {
  "x": 60,
  "y": 503.07692307692304,
}
`;

exports[`Zone grid coordinates converted to a physical location correctly 2`] = `
Object {
  "x": 96.92307692307692,
  "y": 466.15384615384613,
}
`;

exports[`Zone grid coordinates converted to a physical location correctly 3`] = `
Object {
  "x": 503.07692307692304,
  "y": 60,
}
`;

exports[`Zone grid coordinates converted to a physical location correctly 4`] = `
Object {
  "x": 170.76923076923077,
  "y": 318.46153846153845,
}
`;

exports[`Zone grid coordinates converted to a physical location correctly 5`] = `
Object {
  "x": 60,
  "y": 626.6666666666666,
}
`;

exports[`Zone grid coordinates converted to a physical location correctly 6`] = `
Object {
  "x": 94.54545454545455,
  "y": 513.3333333333333,
}
`;

exports[`Zone grid coordinates converted to a physical location correctly 7`] = `
Object {
  "x": 163.63636363636363,
  "y": 60,
}
`;

exports[`Zone grid coordinates converted to a physical location correctly. Error handling. 1`] = `
Object {
  "x": 60,
  "y": 318.46153846153845,
}
`;

exports[`Zone grid coordinates converted to a physical location correctly. Error handling. 2`] = `
Object {
  "x": 405.4545454545455,
  "y": 60,
}
`;

exports[`Zone grid coordinates converted to a physical location correctly. Error handling. 3`] = `
Object {
  "x": 60,
  "y": 60,
}
`;
